src/coreclr/vm/method.cpp

```diff
@@ -3035,6 +3035,9 @@ bool MethodDesc::DetermineAndSetIsEligibleForTieredCompilation()
         // Functional requirement - These methods have no IL that could be optimized
         !IsWrapperStub() &&
 
+        // Functions with AggressiveOptimization attribute bypass tiering
+        !RequestedAggressiveOptimization() &&
+
         // Policy - Generating optimized code is not disabled
         !IsJitOptimizationDisabledForSpecificMethod())
```

Comment:
Each `RequestedAggressiveOptimization`and `IsJitOptimizationDisabledForSpecificMethod` will call `GetImplAttrs`. `GetImplAttrs` is somewhat expensive - it reads raw metadata. 

Can we move this to `IsJitOptimizationDisabledForSpecificMethod` and make sure that `GetImplAttrs` is called just once here?

Reply:
Ah, good point, addressed